### ISSUE TITLE
GitHub action -- grant access to issue queue tags

### ISSUE THREAD

**Issue by jenlampton** [backdrop-ops/backdropcms.org#804]
Once upon a time we had a bot that noticed if you added an issue or a comment in the backdrop-issues queue, and automatically added you to the [Documentation and Organization team](https://github.com/orgs/backdrop/teams/documentation-and-organization) so that you could add labels and milestones to issues in the queue. 

This bot has been out of service for a very long time (IRC GitHub made an API change that made our previous bot non-functional).

I we may be able to accomplish the same thing today by using GitHub actions.

**Comment by BWPanda** [comment]
@jenlampton This page talks about using GH actions for doing some GH-type things (adding labels, moving issues, etc.), but doesn't seem to specifically refer to being able to do things to user accounts...

https://docs.github.com/en/actions/guides/using-github-actions-for-project-management

**Comment by BWPanda** [comment]
This is the closest I could find: https://github.com/marketplace/actions/github-team-sync
It allows you to make changes to a text file, which will then result in changes to your GH teams (e.g. add new teams, rename, add/remove members, etc.).

I don't think this is what you're looking for, but I put it here to show that a GH action can be written to add members to a team (check out that action's source code as an example: https://github.com/rmacklin/team-sync/blob/796e6d4d636bccbcc1f597b5e87b7c56e976d753/src/main.ts#L171)

**Comment by BWPanda** [comment]
The fact that I couldn't find an existing action that automatically adds members to a team when they comment makes me wonder if it's because this shouldn't be automated (generally you want to have some sort of verification process in place). In our case, we want anyone commenting to be able to edit labels, but if the only criteria is to comment, how's that any better/more secure than just allowing logged in user to edit labels...?

In other words, maybe we should just change things from using a group to allowing anyone to edit labels and avoid this issue altogether.

**Comment by jenlampton** [comment]
> how's that any better/more secure than just allowing logged in user to edit labels...?

Is that even possible? ....checking...

**Comment by jenlampton** [comment]
> but if the only criteria is to comment, how's that any better/more secure than just allowing logged in user to edit labels...?

When we set it up, it was "better" because it was possible. I don't think we ever considered it being more secure. 

I just did a quick check and as far as I can tell it's not possible to give every github user write access to a repository.

**Comment by jenlampton** [comment]
Today I discovered that only `Organization owners` are allowed to invite new `members` to an organization, and only existing `members` are allowed to be added to` teams`. This means that there is a very limited number of people who are authorized to manually invite someone to the "Documentation and organization" team (to manage issue labels, assignments, and milestones).

I did discover this github action that will send an invitation to become a member of the organization: https://github.com/ActionsDesk/invite-user

Perhaps we can use that, in combination with 
https://github.com/marketplace/actions/github-team-sync

To first invite a new member, and then add them to the [Documentation and Organization team](https://github.com/orgs/backdrop/teams/documentation-and-organization) team?

The user interface for adding someone to a team allows us to invite them to become an organization member at the same time, so I don't expect there would be an issue with adding someone to a team before the invite was accepted.
